chg: Updated for compatibility with FAB 4.3.10

om0wt opened the pull request:
Includes have been updated to reflect current Flask-Appbuilder (4.3.10) naming convention.

Commit:
chg: Updated for compatibility with FAB 4.3.10

dpgaspar commented on fab_addon_audit/views.py:
```diff
@@ -1,10 +1,12 @@
-from flask.ext.appbuilder import ModelView
-from flask.ext.appbuilder.charts.views import DirectByChartView, GroupByChartView
-from flask.ext.appbuilder.models.group import aggregate_count, aggregate_avg, aggregate_sum
+#from flask.ext.appbuilder.models.sqla.interface import SQLAInterface
```
let's just remove these

dpgaspar reviewed (commented):
Thank you, just removed the commented lines

Commit:
fix: Removed commented includes, fixed chartview

Removed unused includes and fixed issue 922
related error in AuditLogChartView

om0wt commented:
Commented out includes removed and fixed issue 922 related error in AuditLogChartView

Commit:
new: Former target item values comparision table

New and former target item values comparision
table is stored to 'target_values' column now for
updated items. For added and deleted items,
current data are stored to 'target_values'.
Target_values table is shown in AuditLog view.

Commit:
fix: Deleted data fix

EnriqueGautoSand commented:
> Includes have been updated to reflect current Flask-Appbuilder (4.3.10) naming convention.

target values appears as text it didn´t make a table
```
<html>
<body>
<!--StartFragment-->
table class="table table-bordered table-hover"><tr><th>
--


<!--EndFragment-->
</body>
</html>
```

also i recommended
```
class AuditLogView(ModelView):
    datamodel = SQLAInterface(AuditLog)
    base_order = ("created_on", "desc")
    list_widget = ListLinkWidget
    list_columns = ['created_on', 'username', 'operation.name', 'target', 'message']
    base_permissions = ['can_list', 'can_show']
    label_columns = {'showTargetvalues':'Target Values','showOperationName':'Operation'}
    show_fields = ['operation.name', 'message', "username", 'created_on', 'target', 'showTargetvalues']
    show_columns = ['showOperationName','message','username', 'created_on', 'target', 'showTargetvalues']
    
```
and AuditLog model to 

```
class AuditLog(Model):
    id = Column(Integer, Sequence('audit_log_id_seq'), primary_key=True)
    message = Column(String(300), nullable=False)
    username = Column(String(64),  nullable=False)
    created_on = Column(DateTime, default=datetime.datetime.now, nullable=True)
    operation_id = Column(Integer, ForeignKey('operation.id'), nullable=False)
    operation = relationship("Operation")
    target = Column(String(150), nullable=False)
    target_values = Column(String(100000), nullable=True)
    def showTargetvalues(self):
        return Markup(self.target_values)
    def showOperationName(self):
        return self.operation.name
```

EnriqueGautoSand commented:
> Includes have been updated to reflect current Flask-Appbuilder (4.3.10) naming convention.
the update didn´t works

a couple of years ago i changed the modelview to work with update
```
    def pre_pre_update(self, item):
        self.preitem=item
        print('prepre update',item)
        return self.preitem

    def _edit(self, pk):
        """
            Edit function logic, override to implement different logic
            returns Edit widget and related list or None
        """
        is_valid_form = True
        pages = get_page_args()
        page_sizes = get_page_size_args()
        orders = get_order_args()
        get_filter_args(self._filters)
        exclude_cols = self._filters.get_relation_cols()

        item = self.datamodel.get(pk, self._base_filters)
        if not item:
            abort(404)
        self.pre_pre_update(copy.copy(item))
        # convert pk to correct type, if pk is non string type.
        pk = self.datamodel.get_pk_value(item)

        if request.method == "POST":
            form = self.edit_form.refresh(request.form)
            # fill the form with the suppressed cols, generated from exclude_cols
            self._fill_form_exclude_cols(exclude_cols, form)
            # trick to pass unique validation
            form._id = pk
            if form.validate():
                self.process_form(form, False)

                try:
                    form.populate_obj(item)
                    self.pre_update(item)
                except Exception as e:
                    flash(str(e), "danger")
                else:
                    if self.datamodel.edit(item):
                        self.post_update(item)
                    flash(*self.datamodel.message)
                finally:
                    return None
            else:
                is_valid_form = False
        else:
            # Only force form refresh for select cascade events
            form = self.edit_form.refresh(obj=item)
            # Perform additional actions to pre-fill the edit form.
            self.prefill_form(form, pk)

        widgets = self._get_edit_widget(form=form, exclude_cols=exclude_cols)
        widgets = self._get_related_views_widgets(
            item,
            filters={},
            orders=orders,
            pages=pages,
            page_sizes=page_sizes,
            widgets=widgets,
        )
        if is_valid_form:
            self.update_redirect()
        return widgets
```

the pre_update never works so you need to do add a pre_pre_update function, i make it a couple of years ago, i don´t now if still works to catch the update
